lib/AL_USDMaya/AL/usdmaya/nodes/ProxyShapeSelection.cpp

```diff
@@ -1044,6 +1044,20 @@ bool ProxyShape::doSelect(SelectionUndoHelper& helper, const SdfPathVector& orde
   }
   MStringArray newlySelectedPaths;
 
+  auto removeFromMSel = [](MSelectionList& sel, const MObject& toRemove) -> bool {
+    for(uint32_t i = 0; i < sel.length(); ++i)
```

Comment:
Take a copy of sel.length() - it's a dso exported method, so will always be re-evaluated.

Reply:
done!